sgtool/file.go

```diff
@@ -189,11 +191,19 @@ func WithSkipIfFileExists(filepath string) Opt {
 	}
 }
 
-func downloadBinary(ctx context.Context, url string) (io.ReadCloser, func(), error) {
+func WithHTTPHeader(header http.Header) Opt {
```

Comment:
Would it make sense to have this as a key value and then append to the already allocated f.httpHeader? Then you can pass as many opts as you want :)

This option will overwrite any previous headers :)

Reply:
Just so I understand you correctly, do you mean passing each key value pair individually rather than the full map?
i.e. something like
```golang
.WithHTTPHeader("k", "v")
.WithHTTPHeader("k", "v")
...
```

Reply:
Yep exactly :+1: 

You don't need to change, just consider it and see how you feel about it :)

Reply:
Cool! I like it, I don't have a strong preference but I think its neater if you only need to pass small number of headers (which probably is the case most of the time) :)

Reply:
Yeah, if you want you can have both options I guess :shrug: 
`WithHTTPHeaders(<map>)` and `WithHTTPHeader(key,val)` 

But I think the single header use-case is probably going to be the the most used one